adapters/adhese/adhese.go

```diff
@@ -0,0 +1,247 @@
+package adhese
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"sort"
+	"strconv"
+	"strings"
+	"text/template"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/macros"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type AdheseAdapter struct {
+	http             *adapters.HTTPAdapter
+	endpointTemplate template.Template
+}
+
+func (a *AdheseAdapter) Name() string {
+	return "adhese"
+}
+
+func (a *AdheseAdapter) SkipNoCookies() bool {
+	return false
+}
+
+func extractSlotParameter(parameters openrtb_ext.ExtImpAdhese) string {
+	return fmt.Sprintf("/sl%s-%s", parameters.Location, parameters.Format)
```

Comment:
`parameters.Location` and `parameters.Format` are both strings. Please escape them with `url.PathEscape(x)`.